src/lib.rs

```diff
@@ -1,14 +1,230 @@
-pub fn add(left: usize, right: usize) -> usize {
-    left + right
+use ads1x1x::{
+    channel,
+    ic::{Ads1115, Resolution16Bit},
+    interface::I2cInterface,
+    Ads1x1x, SlaveAddr,
+};
+use ak09915_rs::{Ak09915, Mode};
+use bmp280::{Bmp280, Bmp280Builder};
+use embedded_hal::adc::OneShot; //used on ads
+use icm20689::{self, AccelRange, Builder, GyroRange, SpiInterface, ICM20689};
+use linux_embedded_hal::spidev::{self, SpidevOptions};
+use linux_embedded_hal::sysfs_gpio::Direction;
+use linux_embedded_hal::I2cdev;
+use linux_embedded_hal::{Delay, Pin, Spidev};
+use nb::block; //used on ads
+use pwm_pca9685::{Address, Channel, Pca9685};
+
+// Define a struct to hold the sensor data
+pub struct SensorData {
+    pub adc: [i16; 4],
+    pub temperature: f32,
+    pub pressure: f32,
+    pub accelerometer: [f32; 3],
+    pub magnetometer: [f32; 3],
+    pub gyro: [f32; 3],
+}
+
+// Define a struct for the navigator module
+pub struct Navigator {
+    pwm: Pca9685<I2cdev>, //: Pca9685<I2cdev>,
+    bmp: Bmp280,
+    adc: Ads1x1x<I2cInterface<I2cdev>, Ads1115, Resolution16Bit, ads1x1x::mode::OneShot>,
+    imu: ICM20689<SpiInterface<Spidev, Pin>>,
+    mag: Ak09915<I2cdev>,
+    led: Led,
+}
+
+struct Led {
+    first: Pin,
+    second: Pin,
+    third: Pin,
+}
+
```

Comment:
implement to help with access:
```rust
impl Led {
    fn as_mut_array(&self) -> [&mut Pin; 3] {
        &mut led.first, &mut led.second, &mut led.third
    }
}
```

Reply:
Ok, on line 58, it simplifyied the other functions all_on, all_off.

Reply:
Although it's probably going to be useful, I wouldn't mind adding accessors until we write an application that really needs them.